Assets/_Scripts/3Managers/BaseBuildingValidator.cs

```diff
@@ -1,29 +1,78 @@
 ﻿using OrcaDefense.Models;
 using System;
+using System.Collections.Generic;
 
 public class BaseBuildingValidator : IBuildingValidator
 {
-    public BaseBuildingValidator()
+    protected readonly Tile _tile;
+    public BaseBuildingValidator(Tile tile)
     {
+        _tile = tile;
     }
 
-    public virtual bool ShouldRenderRoof(Island map, int x, int y)
+    //Should this building render a roof?
+    public virtual bool ShouldRenderRoof(Island map)
     {
-        return !map.GetTileAt(x, y + 1).IsOccupied;
+        return map.Up(_tile.X, _tile.Y).Building is null;
     }
 
-    public virtual string[] ValidateAdjacencies(Island map, int x, int y)
+    //Does this building care if we build it next to it's neighbours?
+    public virtual string[] ValidateAdjacencies(Island map)
     {
-        return null;
+        return new string[] { };
     }
 
+    //Does this building care if we build X building next to it?
     public virtual string[] ValidateAdjacency(BuildingTypeEnum buildingToBuild)
     {
-        return null;
+        return new string[] { };
     }
 
-    public virtual string[] ValidateCanBuildOnTop(Island map, int x, int y)
+    //Can the building be placed in this position?
+    public virtual string[] ValidatePosition(Island map)
     {
-        return null;
+        //IsOccupied
+        var errors = new List<string>();
+        if (_tile.Building != null) { errors.Add("There's already a building there!"); }
+
+        //IsSupported
+        if (_tile.Y != 0 && map.Down(_tile.X, _tile.Y).Building is null) { errors.Add("Tile is not supported!"); }
+
+        return errors.ToArray();
+    }
+
+    //Can the given building be built on top?
```

Comment:
Shouldn't this method be for determining whether you can build another building on top of this one?

Reply:
Skip for now